platform/app/src/routes/Mode/Mode.tsx

```diff
@@ -25,14 +25,45 @@ const { getSplitParam } = utils;
  * @returns array of subscriptions to cancel
  */
 function defaultRouteInit(
-  { servicesManager, studyInstanceUIDs, dataSource, filters },
+  { servicesManager, studyInstanceUIDs, dataSource, filters, appConfig },
   hangingProtocolId
 ) {
   const {
     displaySetService,
     hangingProtocolService,
     uiNotificationService,
   } = servicesManager.services;
+  const minDisplaySetsToRunHP = appConfig?.minDisplaySetsToRunHP;
+
+  /**
+   * Function to apply the hanging protocol when the minimum number of display sets were
+   * received or all display sets retrieval were completed
+   * @returns
+   */
+  function applyHangingProtocol() {
+    // check if a hanging protocol is already applied. If so abort execution
+    if (hangingProtocolService.getActiveProtocol().protocol) {
+      return;
```

Comment:
this doesn't make sure it is already applied. it means that it only exists

Reply:
The protocol attribute inside the hangingProtocolService object is assigned in three ocasions: in the creation of the object, in reset method, which is called whenever OHIF enters a new mode, and in the setProtocol method, which has a try/catch block that returns to the previous state in case of a problem. So if the protocol attribute is assigned it means someone called the setProtocol function. So if it is assigned and not applied it is an internal state error, that could happen without this code, right?